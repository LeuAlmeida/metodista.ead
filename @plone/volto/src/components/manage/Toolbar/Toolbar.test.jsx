import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-intl-redux';
import jwt from 'jsonwebtoken';

import Toolbar from './Toolbar';

const mockStore = configureStore();

describe('Toolbar', () => {
  it('renders an actions component', () => {
    const store = mockStore({
      userSession: {
        token: jwt.sign({ fullname: 'John Doe' }, 'secret'),
      },
      content: {
        data: {
          '@type': 'Folder',
          is_folderish: true,
        },
      },
      intl: {
        locale: 'en',
        messages: {},
      },
    });
    const component = renderer.create(
      <Provider store={store}>
        <Toolbar pathname="/test" inner={<span />} />
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
