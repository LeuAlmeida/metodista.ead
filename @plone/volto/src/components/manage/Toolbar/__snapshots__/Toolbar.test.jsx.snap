// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar renders an actions component 1`] = `
Array [
  <div
    className="ui borderless vertical icon left fixed menu"
  >
    <span />
    <div
      className="item logo"
      onClick={[Function]}
    >
      <div
        className="ui divider"
      />
      <div
        className="image"
        style={
          Object {
            "backgroundImage": "url(pastanaga.svg)",
          }
        }
      />
    </div>
    <button
      className="ui button trigger"
      onClick={[Function]}
    />
  </div>,
  <div
    className="pusher expanded"
  />,
]
`;
